### DISCUSSION TITLE
FR: Allow Asset Volumes To Behave Like Structures

### DISCUSSION THREAD

**Discussion by RitterKnightCreative** [craftcms/cms#8550] [Ideas]
For the most part, Assets in Craft are first class citizens. 

But it'd be nice for a volume to be able to control the **sort order** (and hierarchy) of assets in certain situations and behave like structures. 

We have use case where a client has multiple assets that relate back to multiple companies (in this case pages). Those assets could live on multiple pages and it'd be nice to be able to control the sort order in one spot. 

The kicker is one of the people at the company also doesn't necessary need to have access to any other resource on the site than the assets themselves.

**Alternatives:**
* Set a sort order field (or multiple titles) in the asset.  **Downside:** Not very visual.
* Setup multiple pages and then bring in the assets for each page (eg assets field, rich text, matrix, etc.) **Downside:** overly complex, not necessary, have to open up a new section/permissions for the author. 

It would also be interesting to be able to nest assets, like a table of contents. I could envision that being useful in some situations as well.

**Comment by Saboteur777** [comment]
I recently run into this as an issue, and I agree Asset Volumes should be able to behave like structures. 

Creating a separate structure section for these items could be a workaround (a structure section where a field relates to the asset), but it seems counterintuitive: Assets are Elements already, why create another section with Elements and double the work for content editors? 

Another workaround could be to add a numeric field to assets, but that's just... bad. :\ 

If the current behaviour is expected, we could still use that by setting `orderBy` in the query for the frontend or in the Control Panel.

**Comment by alexanderloewe** [comment]
I like that idea.
